bmeg_app/__main__.py

```diff
@@ -0,0 +1,153 @@
+            id="blurb",
+        ),
+        dbc.Collapse(
+            dbc.Nav(
+                [
+                    dbc.NavLink("Compare Drug Treatments", href="/page-1", id="page-1-link",
+                        style={'font-size':styles['t']['size_font_card'],
+                        'fontFamily':styles['t']['type_font']
+                        },),
+                    dbc.NavLink("TCGA Clustering", href="/page-2", id="page-2-link",
+                        style={'font-size':styles['t']['size_font_card'],
+                        'fontFamily':styles['t']['type_font']
+                        },),
+                    dbc.NavLink("Literature Gene-Drug Associations", href="/page-3", id="page-3-link",
+                        style={'font-size':styles['t']['size_font_card'],
+                        'fontFamily':styles['t']['type_font']
+                        },),
+                    dbc.NavLink("BMEG Data Types", href="/page-4", id="page-4-link",
+                        style={'font-size':styles['t']['size_font_card'],
+                        'fontFamily':styles['t']['type_font']
+                        },),
+                ],
+                vertical=True,
+                pills=True,
+            ),
+            id="collapse",
+        ),
+    ],
+    id="sidebar",
+)
+content = html.Div(id="page-content")
+app.layout = html.Div([dcc.Location(id="url"), sidebar, content])
+
+
+@app.callback(
+    [Output(f"page-{i}-link", "active") for i in range(1, 5)],
+    [Input("url", "pathname")],
+)
+def toggle_active_links(pathname):
+    if pathname == "/":
+        return True, False, False, False
+    return [pathname == f"/page-{i}" for i in range(1, 5)]
+
+@app.callback(
+    Output("page-content", "children"), 
+    [Input("url", "pathname")]
+)
+def render_page_content(pathname):
+    if pathname in ["/", "/page-1"]:
+        return html.Div([compare_dresp_view.tab_layout]) 
+    elif pathname == "/page-2":
+        return html.Div([tumor_match_normal_view.tab_layout])    
+    elif pathname == "/page-3":
+        return html.Div([literature_support_view.tab_layout]) 
+    elif pathname == "/page-4":
+        return html.Div([data_types_view.tab_layout])
+    return dbc.Jumbotron(
+        [
+            html.H1("404: Not found", className="text-danger"),
+            html.Hr(),
+            html.P(f"The pathname {pathname} was not recognised..."),
+        ]
+    )
+
+@app.callback(
+    Output("sidebar", "className"),
+    [Input("sidebar-toggle", "n_clicks")],
+    [State("sidebar", "className")]
+)
+def toggle_classname(n, classname):
+    if n and classname == "":
+        return "collapsed"
+    return ""
+
+@app.callback(
+    Output("collapse", "is_open"),
+    [Input("navbar-toggle", "n_clicks")],
+    [State("collapse", "is_open")]
+)
+def toggle_collapse(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+if __name__ == '__main__':
+    app.run_server(host='0.0.0.0', debug=True, port=80)
```

Comment:
Should implement a command line parser, so the users can pass the 'debug' and 'port' values can be configured on the command line (I always end up manually setting the port to 8080 when testing)

Reply:
Config file added in commit 
ff17906